src/Builder.jsx

```diff
@@ -497,7 +543,16 @@ class Builder {
           {...this.map}
           mode={mode}
           settings={this.settings}
-          saveMap={() => this.map.save()}
+          saveMap={() => {
+            if (this.semanticOptions) {
```

Comment:
can you comment on the purpose of this logic?

Reply:
Added a comment explaining why I did it but basically if semanticZoom is active it reverts the options to their original loaded values before saving